server/neptune/workflows/internal/github/link/builder.go

```diff
@@ -0,0 +1,30 @@
+package link
+
+import (
+	"fmt"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/root"
+	"net/url"
+	"path"
+)
+
+type Builder struct{}
+
+// BuildDownloadLinkFromArchive is a helper fxn that isolates the logic of modifying a GH archive link
+// into source url that the go-getter library understand for downloading
+func (b Builder) BuildDownloadLinkFromArchive(archiveURL *url.URL, root root.Root, repo github.Repo) string {
+	// Add archive query parameter for getter library to extract archive
+	queryParams := "archive=zip"
+	token := archiveURL.Query().Get("token")
+	if token != "" {
+		queryParams += fmt.Sprintf("&token=%s", token)
+	}
```

Comment:
when would this be empty or not empty?

Reply:
Public vs. private repository I believe (at least from some testing). The GH docs are pretty vague on this: https://docs.github.com/en/rest/repos/contents#download-a-repository-archive-zip